controllers/applicationvolumereplication_controller.go

```diff
@@ -189,9 +202,15 @@ func (r *ApplicationVolumeReplicationReconciler) createVRGManifestWork(
 		return empty, empty, fmt.Errorf("no peer cluster found for subscription %s", subscription.Name)
 	}
 
-	if err := r.doCreateOrUpdateManifestWork(
+	if err := r.createOrUpdateVRGRolesManifestWork(homeCluster); err != nil {
+		r.Log.Error(err, "Failed to create or update VolumeReplicationGroup Roles manifest")
+
+		return empty, empty, err
+	}
+
+	if err := r.createOrUpdateVRGManifestWork(
 		subscription.Name, subscription.Namespace, homeCluster); err != nil {
-		r.Log.Error(err, "failed to create or update manifest")
+		r.Log.Error(err, "Failed to create or update VolumeReplicationGroup manifest")
```

Comment:
error log message should start in lowercase per: https://github.com/golang/go/wiki/CodeReviewComments#error-strings